samples/navigate-route/src/main/java/com/esri/arcgismaps/sample/navigateroute/components/NavigateRouteViewModel.kt

```diff
@@ -0,0 +1,417 @@
+        // get the route's geometry from the route result
+        val routeGeometry: Polyline = routeResult.routes[0].routeGeometry
+            ?: return messageDialogVM.showMessageDialog("Route is missing geometry")
+
+        // set up a simulated location data source which simulates movement along the route
+        val simulationParameters = SimulationParameters(
+            Instant.now(),
+            velocity = 35.0,
+            horizontalAccuracy = 5.0,
+            verticalAccuracy = 5.0
+        )
+
+        // create the simulated data source using the geometry and parameters
+        val simulatedLocationDataSource = SimulatedLocationDataSource(
+            polyline = routeGeometry,
+            parameters = simulationParameters
+        )
+
+        // set up a RouteTracker for navigation along the calculated route
+        val routeTracker = RouteTracker(
+            routeResult = routeResult,
+            routeIndex = 0,
+            skipCoincidentStops = true
+        ).apply {
+            setSpeechEngineReadyCallback {
+                isTextToSpeechInitialized.get() && textToSpeech?.isSpeaking == false
+            }
+        }
+
+
+        // create a route tracker location data source to snap the location display to the route
+        val routeTrackerLocationDataSource = RouteTrackerLocationDataSource(
+            routeTracker = routeTracker,
+            locationDataSource = simulatedLocationDataSource
+        )
+
+        locationDisplayJob = with(viewModelScope) {
+            launch(Dispatchers.IO) {
+                // automatically enable recenter button when navigation pan is disabled
+                locationDisplay.autoPanMode.filter { it == LocationDisplayAutoPanMode.Off }
+                    .collect { isRecenterButtonEnabled = true }
+            }
+
+            launch(Dispatchers.IO) {
+                // set the simulated location data source as the location data source for this app
+                locationDisplay.dataSource = routeTrackerLocationDataSource
+
+                // start the location data source
+                locationDisplay.dataSource.start().getOrElse {
+                    messageDialogVM.showMessageDialog(
+                        title = "Error starting location data source",
+                        description = it.message.toString()
+                    )
+                }
+
+                // set the auto pan to navigation mode
+                locationDisplay.setAutoPanMode(LocationDisplayAutoPanMode.Navigation)
+
+                // data source has started, display stop message
+                nextStopText = getStringArray(R.array.stop_message)[0]
+
+                // plays the direction voice guidance
+                updateVoiceGuidance(routeTracker)
+
+                launch(Dispatchers.IO) {
+                    // zoom in the scale to focus on the navigation route
+                    // TODO This seems to get absorbed by LocationDisplayAutoPanMode
```

Comment:
There's a TODO here

Reply:
Ah yes, this was added since I could not get both the zooming scale and auto pan mode to navigation to work simultaneously when the user starts to navigate. I added more about this in the PR description. I tried to make something similar along the lines of a typical navigation app where the viewpoint zooms down to the street level and follows the location display, and I am trying to understand if this "effect" is/is not possible using the SDK. 

Weirdly, when you click recenter while navigating the viewpoint scales down to `10000.0` but recenter logic only sets the `setAutoPanMode`. You might need to switch between "Reset" -> "Navigate" a few times to see it.